index.js

```diff
@@ -275,8 +275,18 @@ Router.prototype._on = function _on (method, path, opts, handler, store) {
   }
 
   for (const existRoute of this._routesPatterns) {
+    let samePath = false
+
+    if (existRoute.path === path) {
+      samePath = true
+    } else if (existRoute.path === '/*' && path === '*') {
```

Comment:
can't this be optimized by skipping both checks altogether if path is neither? (which we know even priir to entering cycle)